From 08c50aa9b063b448584e94ab3033cb44cb70fdbd Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@freescale.com>
Date: Mon, 8 Jun 2015 15:55:04 +0800
Subject: [PATCH] MLK-11053: ASoC: imx_mqs: Remove 96k and 192k support for
 mqs

If the mclk is 24.576MHz, mqs can't support 96k and 192kHz, because
the we can't get a proper clock divider for mqs.

Signed-off-by: Shengjiu Wang <shengjiu.wang@freescale.com>
---
 sound/soc/fsl/imx-mqs.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/sound/soc/fsl/imx-mqs.c b/sound/soc/fsl/imx-mqs.c
index 85ad252..4e32e42 100644
--- a/sound/soc/fsl/imx-mqs.c
+++ b/sound/soc/fsl/imx-mqs.c
@@ -40,10 +40,8 @@ static int imx_mqs_startup(struct snd_pcm_substream *substream)
 
 	if (priv->mclk_freq == 24576000) {
 		support_rates[0] = 48000;
-		support_rates[1] = 96000;
-		support_rates[2] = 192000;
 		constraint_rates.list = support_rates;
-		constraint_rates.count = 3;
+		constraint_rates.count = 1;
 
 		ret = snd_pcm_hw_constraint_list(runtime, 0, SNDRV_PCM_HW_PARAM_RATE,
 							&constraint_rates);
-- 
1.8.0

